keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/DoubleCPTVertex.java

```diff
@@ -3,17 +3,52 @@
 import io.improbable.keanu.tensor.Tensor;
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
 import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.Differentiable;
 import io.improbable.keanu.vertices.dbl.DoubleVertex;
-import io.improbable.keanu.vertices.generic.nonprobabilistic.CPTVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
+import io.improbable.keanu.vertices.generic.nonprobabilistic.CPTCondition;
+import io.improbable.keanu.vertices.update.NonProbabilisticValueUpdater;
 
 import java.util.List;
 import java.util.Map;
 
-public class DoubleCPTVertex extends CPTVertex<DoubleTensor> {
+public class DoubleCPTVertex extends DoubleVertex implements Differentiable {
+
+    private final List<Vertex<? extends Tensor<Boolean>>> inputs;
+    private final Map<CPTCondition, DoubleVertex> conditions;
+    private final DoubleVertex defaultResult;
 
     public DoubleCPTVertex(List<Vertex<? extends Tensor<Boolean>>> inputs,
-                           Map<Condition, DoubleVertex> conditions,
+                           Map<CPTCondition, DoubleVertex> conditions,
                            DoubleVertex defaultResult) {
-        super(inputs, conditions, defaultResult);
+        super(new NonProbabilisticValueUpdater<>(v -> ((DoubleCPTVertex) v).op()));
```

Comment:
In this case it will always be `this` which can safely be cast to the `this` class. That being said, I don't like it either. The need for the value updater has been removed in another PR I have up at the moment: https://github.com/improbable-research/keanu/pull/165